tests/test_urldispatch.py

```diff
@@ -774,7 +774,21 @@ async def test_dynamic_match_unquoted_path(router: web.UrlDispatcher) -> None:
     resource_id = "my%2Fpath%7Cwith%21some%25strange%24characters"
     req = make_mocked_request("GET", f"/path/{resource_id}")
     match_info = await router.resolve(req)
-    assert match_info == {"path": "path", "subpath": unquote(resource_id)}
+    assert match_info == {
+        "path": "path",
+        "subpath": "my%2Fpath|with!some%strange$characters",
```

Comment:
If we don't unquote `%2F` everything works downstream

It seems we already accepted that as OK...
https://github.com/aio-libs/aiohttp/blob/203ca4273edcd6ea5372c630b054381ebb3bacdb/tests/test_urldispatch.py#L489
https://github.com/aio-libs/aiohttp/blob/203ca4273edcd6ea5372c630b054381ebb3bacdb/tests/test_urldispatch.py#L667

...but should we be reverting this and the yarl change instead since this change is in a `.6` patch release?

Reply:
Those 2 tests are using url_for(), i.e. the URL needed to request those files. If it unquoted the %2F, then the URL would be wrong. I don't think that's related to what the matchinfo should be.

I guess the problem is that the handler for an endpoint with such a url may be written like:
`for parts in request.match_info["filepath"].split("/")`

If we don't decode the `%2F`, users will need to do the decoding now (or split on that), but all we've really done if offload the unsolvable issue onto them. Because, now the user won't be able to tell the difference between `%2F` and `%252F`.

Reply:
With the change I just pushed along with https://github.com/aio-libs/yarl/pull/1057 these tests go back to the original behavior before https://github.com/aio-libs/aiohttp/pull/8898 of decoding `%2F`